File: test/ViewController.m
//
//  ViewController.m
//  test
//
//  Created by 宛越 on 16/6/17.
//  Copyright © 2016年 yuewan. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)test:(id)sender {
    //NSViewController* viewController = [[NSViewController alloc] init];
    
    NSStoryboard * sb = [NSStoryboard storyboardWithName:@"Main" bundle:nil];
    NSViewController* viewController = [sb instantiateControllerWithIdentifier:@"viewCon"];
    viewController.view.bounds = CGRectMake(0, 0, 800, 600);
    
    
//    [self presentViewControllerAsModalWindow:viewController];
    [self presentViewController:viewController asPopoverRelativeToRect:self.view.frame ofView:self.view preferredEdge:NSMinXEdge behavior:NSPopoverBehaviorApplicationDefined];

}

- (void)viewDidLoad {
    [super viewDidLoad];

    // Do any additional setup after loading the view.
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

@end

File: test/ViewController.swift
//
//  ViewController.swift
//  test
//
//  Created by 宛越 on 16/6/20.
//  Copyright © 2016年 yuewan. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    @IBAction func touch(sender: AnyObject) {
        
//        NSStoryboard * sb = [NSStoryboard storyboardWithName:@"Main" bundle:nil];
//        MainWindowController* mainWindowCont = [sb instantiateControllerWithIdentifier:@"mainWindowController"];

        
        let sb = NSStoryboard(name: "Main", bundle: nil);
        let mainWin = sb.instantiateControllerWithIdentifier("mainWindowController") as! MainWindowController;
        
//        NSApp.
        mainWin.showWindow(nil);
        
        if mainWin.window == nil {
            print("error");
        }
        
        
        //presentViewControllerAsModalWindow(mainWin);
//        mainWin.showWindow(self);
//        mainWin.window
        //mainWin.showWindow(nil)
        //mainWin.window?.makeMainWindow()
    }
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override var representedObject: AnyObject? {
        didSet {
        // Update the view, if already loaded.
        }
    }


}

